int32[17] word
int32 row
int32 column
bool horizontal
